### PR TITLE
fix: not running all requests in the Postman collection

### LINKED ISSUE #730: Not running all requests in the Postman collection
See the discussion #719

### PR DESCRIPTION
**Pull request by kevin-hyperjump**
# Monika Pull Request (PR)  

1. Fixes #730
2. Adds support for Postman collection `v2.0`

### CODE REVIEW

**Comment by kevin-hyperjump** [comment]
## Example of Postman collection v2.0

```json
{
  "info": {
    "_postman_id": "0653fc29-a517-4f78-820a-66b207c7cd62",
    "name": "Monika Monitor Collection",
    "schema": "https://schema.getpostman.com/json/collection/v2.0.0/collection.json"
  },
  "item": [
    {
      "name": "[GitHub API] Hyperjump",
      "protocolProfileBehavior": {
        "disableBodyPruning": true
      },
      "request": {
        "method": "GET",
        "header": [],
        "body": {
          "mode": "raw",
          "raw": "{\n    \"any\": \"any\"\n}\n",
          "options": {
            "raw": {
              "language": "json"
            }
          }
        },
        "url": "https://api.github.com/users/hyperjumptech"
      },
      "response": []
    },
    {
      "name": "[GitHub API] Hyperjump Repos",
      "request": {
        "method": "GET",
        "header": [],
        "url": "https://api.github.com/users/hyperjumptech/repos"
      },
      "response": []
    }
  ]
}
```

## Example of Postman collection v2.1

```json
{
  "info": {
    "_postman_id": "0653fc29-a517-4f78-820a-66b207c7cd62",
    "name": "Monika Monitor Collection",
    "schema": "https://schema.getpostman.com/json/collection/v2.1.0/collection.json"
  },
  "item": [
    {
      "name": "[GitHub API] Hyperjump",
      "request": {
        "method": "GET",
        "header": [],
        "url": {
          "raw": "https://api.github.com/users/hyperjumptech",
          "protocol": "https",
          "host": [
            "api",
            "github",
            "com"
          ],
          "path": [
            "users",
            "hyperjumptech"
          ]
        }
      },
      "response": []
    },
    {
      "name": "[GitHub API] Hyperjump Repos",
      "request": {
        "method": "GET",
        "header": [],
        "url": {
          "raw": "https://api.github.com/users/hyperjumptech/repos",
          "protocol": "https",
          "host": [
            "api",
            "github",
            "com"
          ],
          "path": [
            "users",
            "hyperjumptech",
            "repos"
          ]
        }
      },
      "response": []
    }
  ]
}
```

File: src/components/config/parse-postman.ts [hunk 1]
```diff
@@ -23,65 +23,75 @@
...
+  if (config?.info?.schema?.includes('v2.0')) {
+    return 'v2.0'
   }
 
-  return probe
+  if (config?.info?.schema?.includes('v2.1')) {
+    return 'v2.1'
+  }
+
+  throw new Error('UnsupportedVersion')
```
Comment (dennypradipta) [inline src/components/config/parse-postman.ts:39]: Seeing this, do we need to add an information to the docs saying that only v2.0 and v2.1 collection are supported?

Comment (kevin-hyperjump) [reply src/components/config/parse-postman.ts:39]: Sure, I'll add the information to the docs.

**Review by nicnocquee — CHANGES_REQUESTED** [review]
Can you also add the test to make sure the parsed postman config contain the expected number of probes? I think right now the test only checks the first probe.

**Comment by kevin-hyperjump** [comment]
> Can you also add the test to make sure the parsed postman config contain the expected number of probes? I think right now the test only checks the first probe.

I've rewritten the entire Postman tests, both integration and unit test 😈

**Review by nicnocquee — APPROVED** [review]

**Review by sapiderman — APPROVED** [review]
:ghost: :alien: :space_invader:

**Comment by rgraham7503** [comment]
I'm getting the same error on a simple postman get v2.0 collection
`{
	"info": {
		"_postman_id": "xxxxx",
		"name": "Monika Monitor Collection",
		"schema": "https://schema.getpostman.com/json/collection/v2.0.0/collection.json",
		"_exporter_id": "xxxxxx"
	},
	"item": [
		{
			"name": "GET Basic status page",
			"request": {
				"method": "GET",
				"header": [
					{
						"key": "Content-Type",
						"value": "application/x-www-form-urlencoded",
						"type": "text"
					}
				],
				"url": "https://xxxxx/api/v2/status.json"
			},
			"response": []
		}
        }
`

**Pull request merged**